python/lsst/ts/gis/component.py

```diff
@@ -64,20 +66,56 @@ async def update_status(self):
         reply = await self.commander.read()
         if reply is not None:
             raw_status = self.commander.get_raw_string(reply)
-            if self.raw_status != raw_status:
-                await self.csc.evt_rawStatus.set_write(status=raw_status)
-                self.log.debug(f"registers={reply.registers}")
-            self.raw_status = raw_status
-            for index, (current_subsystem, old_subsystem) in enumerate(
-                itertools.zip_longest(
-                    reply.registers, self.system_status, fillvalue=None
+            await self.csc.evt_rawStatus.set_write(status=raw_status)
+            self.log.debug(f"registers={reply.registers}")
+            # Since salobj checks the cached data with set_write, only need
+            # to iterate through the current reply.
+            for index, current_subsystem in enumerate(reply.registers):
+                await self.csc.evt_systemStatus.set_write(
+                    index=index, status=current_subsystem
                 )
-            ):
-                if current_subsystem != old_subsystem:
-                    await self.csc.evt_systemStatus.set_write(
-                        index=index, status=current_subsystem
-                    )
-            self.system_status = reply.registers
+            await self.fill_out_fields(raw_status)
+
+    async def fill_out_fields(self, raw_status):
+        """Fill out subsystem event data.
+
+        TBD
+
+        Parameters
+        ----------
+        raw_status: `str`
+            The string array of 1and0's that comprise the status of the GIS's
+            subsystems.
+        """
+        statuses = raw_status.split(" ")
+        for index, status in enumerate(statuses):
+            subsystem_name = getattr(enums, subsystem_order[index])
+            # For a given string of 1 and 0's, return an array of booleans
+            if hasattr(subsystem_name, "tuple_range"):
+                tuple_min, tuple_max = subsystem_name.tuple_range()
+            else:
+                tuple_min = tuple_max = None
+            t = tuple()
+            status2 = []
+            appended_tuple = False
+            for index2, bit_status in enumerate(status):
+                if tuple_max is not None and tuple_min is not None:
+                    if index2 >= tuple_min and index2 < tuple_max:
+                        t += tuple([bool(int(bit_status))])
+                    elif index2 + 1 > tuple_max and not appended_tuple:
+                        t += tuple([bool(int(bit_status))])
+                        status2.append(t)
+                        appended_tuple = True
+                    else:
+                        status2.append(bool(int(bit_status)))
+                else:
+                    status2.append(bool(int(bit_status)))
+            self.log.info(f"{status2=}")
```

Comment:
Please avoid `info` log messages in CSCs. This doesn't add useful info to the logs so set it to debug instead.